### PR TITLE
@jorystiefel => Support Web-to-Native Handoff and Universal Links 📱

### PR DESCRIPTION
**Pull request by alloy**
Part of #767.

### CODE REVIEW

**Comment by alloy** [comment]
Currently waiting for https://github.com/artsy/force/pull/3408

File: Podfile.lock [hunk 1]
```diff
@@ -257,7 +257,7 @@ SPEC CHECKSUMS:
   Adjust: 2c3d4db6047c6bba68d7c5ede63fbdc989eb57b7
   AFNetworkActivityLogger: 121486778117d53b3ab1c61d264b88081d0c3eee
   AFNetworking: 05edc0ac4c4c8cf57bcf4b84be5b0744b6d8e71e
-  AFOAuth1Client: 67c1e4718082d6b244b3f837452eb79f36a48003
+  AFOAuth1Client: 07ccc935ba06ac8d023c16d39a5bdf04bc6f92e1
```
Comment (orta) [inline Podfile.lock]: this all caching stuff?

Comment (orta) [inline Podfile.lock]: just merged #784

Comment (alloy) [inline Podfile.lock]: Ah yeah, I branched from there… :)

**Comment by orta** [comment]
Looks like this gives us the ability to do shared keychains too: https://developer.apple.com/library/ios/documentation/Security/Reference/SharedWebCredentialsRef/

**Comment by alloy** [comment]
@orta Filed a new ticket for that #799.

**Comment by alloy** [comment]
@jorystiefel This should be good to go on green.

File: Artsy/App/ARAppActivityContinuationDelegate.m [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@implementation ARAppActivityContinuationDelegate
+
++ (void)load
+{
+    [JSDecoupledAppDelegate sharedAppDelegate].activityContinuationDelegate = [[self alloc] init];
+}
+
+- (BOOL)application:(UIApplication *)application willContinueUserActivityWithType:(NSString *)userActivityType;
+{
+    return [userActivityType isEqualToString:NSUserActivityTypeBrowsingWeb] || [userActivityType hasPrefix:@"net.artsy."];
```
Comment (alloy) [inline Artsy/App/ARAppActivityContinuationDelegate.m]: @jorystiefel Added support for your activity types.

**Comment by alloy** [comment]
Gonna add one more small patch, just a sec.

File: Artsy_Tests/App_Tests/ARAppActivityContinuationDelegateTests.m [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#import "ARAppActivityContinuationDelegate.h"
+#import "ARTopMenuViewController.h"
+#import "ARArtworkSetViewController.h"
+#import "ARArtworkViewController.h"
+
+SpecBegin(ARAppActivityContinuationDelegate);
```
Comment (orta) [inline Artsy_Tests/App_Tests/ARAppActivityContinuationDelegateTests.m]: niccceeeee - tests for UIAppDelegate

File: Artsy/App/ARAppActivityContinuationDelegate.m [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler;
+{
+    UIViewController *viewController = [ARSwitchBoard.sharedInstance loadURL:userActivity.webpageURL];
+    if (viewController) {
+        [[ARTopMenuViewController sharedController] pushViewController:viewController];
+    }
+    return YES;
+}
+
+//- (void)application:(UIApplication *)application didFailToContinueUserActivityWithType:(NSString *)userActivityType error:(NSError *)error;
```
Comment (orta) [inline Artsy/App/ARAppActivityContinuationDelegate.m]: want to keep these in?

Comment (alloy) [inline Artsy/App/ARAppActivityContinuationDelegate.m]: Good point, maybe I should at least log something, not sure when it could fail

Comment (alloy) [inline Artsy/App/ARAppActivityContinuationDelegate.m]: Read up on them, don’t make sense for us to implement.

Comment (orta) [inline Artsy/App/ARAppActivityContinuationDelegate.m]: :+1:

**Comment by alloy** [comment]
@jorystiefel Good to go :+1:

File: Artsy/Artsy.entitlements [hunk 5]
```diff
@@ -2,6 +2,11 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>com.apple.developer.associated-domains</key>
+	<array>
+		<string>applinks:www.artsy.net</string>
```
Comment (jorystiefel) [inline Artsy/Artsy.entitlements]: @alloy I'm sure there's a reason but I'm not aware: why do we specify www.artsy.net and not artsy.net?

Comment (alloy) [inline Artsy/Artsy.entitlements]: Good point, there isn’t really a naked domain host, so did not get to testing that yet. Will do so now.

Comment (alloy) [inline Artsy/Artsy.entitlements]: I’ve added the naked domain. It won’t work yet, though, but that’s for another ticket.

**Comment by jorystiefel** [comment]
Awesome 🔍

**Pull request merged**